### PR TITLE
Jbaublitz issue project 192

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
unique instance stalled.

**Comment by mulkieran** [comment]
stratisd error is just due to reboot.

**Review by jbaublitz — APPROVED** [review]

**Comment by mulkieran** [comment]
rebased.

File: src/engine/strat_engine/backstore/crypt.rs [hunk 1]
```diff
@@ -599,6 +583,31 @@ impl CryptHandle {
     }
 }
 
+/// Interpret non-Clevis keys that may contain additional information about
+/// how to configure Clevis when binding. Remove any expected non-Clevis keys
+/// from the configuration and return it.
+/// The only value to be returned is whether or not the bind command should be
+/// passed the argument yes, which is returned in the first item of the
+/// returned pair.
+pub fn interpret_clevis_config(pin: &str, mut clevis_config: Value) -> Result<(bool, Value)> {
```
Comment (jbaublitz) [inline src/engine/strat_engine/backstore/crypt.rs:592]: I think a prototype of `&str, &mut Value` would make more sense here, remove the key-value pairs through the mutable reference, and then only return the data that has been parsed out of the `Value`.

**Review by jbaublitz — CHANGES_REQUESTED** [review]

**Review by drckeefe — APPROVED** [review]

File: src/engine/strat_engine/backstore/crypt.rs (hunk 1 above)
Comment (mulkieran) [reply src/engine/strat_engine/backstore/crypt.rs]: > I think a prototype of `&str, &mut Value` would make more sense here, remove the key-value pairs through the mutable reference, and then only return the data that has been parsed out of the `Value`.

Done!

**Review by jbaublitz — APPROVED** [review]

**Comment by mulkieran** [comment]
Superceded by #2314

**Pull request closed without merging**